lockc/src/main.rs

```diff
@@ -0,0 +1,178 @@
+use std::{env, path, thread};
+
+use anyhow::Result;
+use log::{debug, error};
+use simplelog::{ColorChoice, ConfigBuilder, LevelFilter, TermLogger, TerminalMode};
+use thiserror::Error;
+use tokio::{
+    runtime::Runtime,
+    sync::{mpsc, oneshot},
+};
+
+mod bpfstructs;
+mod communication;
+mod load;
+mod maps;
+mod runc;
+mod settings;
+mod sysutils;
+
+use communication::EbpfCommand;
+use load::{attach_programs, load_bpf};
+use maps::{add_container, add_process, delete_container, init_allowed_paths};
+use runc::RuncWatcher;
+use sysutils::check_bpf_lsm_enabled;
+
+#[derive(Error, Debug)]
+enum FanotifyError {
+    #[error("could not send the message")]
+    Send,
+}
+
+/// Runs an fanotify-based runc watcher, which registers containers every time
+/// they are created or deleted.
+fn fanotify(
+    fanotify_bootstrap_rx: oneshot::Receiver<()>,
+    ebpf_tx: mpsc::Sender<EbpfCommand>,
+) -> Result<()> {
+    RuncWatcher::new(fanotify_bootstrap_rx, ebpf_tx)?.work_loop()?;
+    Ok(())
```

Comment:
nitpicking: not needed, use the result from the previous line

Reply:
Same here, I would need to explicitly convert the error from previous line with `map_err`. I prefer staying with `Ok(())`